src/bundle/Resources/public/js/CKEditor/anchor/anchor-editing.js

```diff
@@ -17,6 +17,67 @@ class IbexaAnchorEditing extends Plugin {
                 key: 'id',
             },
         });
+
+        this.editor.conversion.for('dataDowncast').add((dispatcher) => {
+            dispatcher.on('attribute:anchor:listItem', (event, data, conversionApi) => {
+                if (data.attributeKey !== 'anchor') {
+                    return;
+                }
+
+                const viewItem = conversionApi.mapper.toViewElement(data.item);
+                const previousElement = viewItem.parent.previousSibling;
+
+                if (data.attributeNewValue) {
+                    conversionApi.writer.setAttribute('id', data.attributeNewValue, viewItem.parent);
+                } else {
+                    conversionApi.writer.removeAttribute('id', viewItem.parent);
+                }
+
+                conversionApi.writer.removeAttribute('id', viewItem);
+
+                if (previousElement?.name === viewItem.parent.name) {
+                    conversionApi.writer.mergeContainers(conversionApi.writer.createPositionAfter(previousElement));
+                }
+            });
+        });
+
+        this.editor.conversion.for('editingDowncast').add((dispatcher) => {
+            dispatcher.on('attribute:anchor:listItem', (event, data, conversionApi) => {
+                if (data.attributeKey !== 'anchor') {
+                    return;
+                }
+
+                const viewItem = conversionApi.mapper.toViewElement(data.item);
+                const previousElement = viewItem.parent.previousSibling;
+                const nextElement = viewItem.parent.nextSibling;
+
+                if (data.attributeNewValue) {
+                    conversionApi.writer.setAttribute('id', data.attributeNewValue, viewItem.parent);
+                } else {
+                    conversionApi.writer.removeAttribute('id', viewItem.parent);
+                }
+
+                conversionApi.writer.removeAttribute('id', viewItem);
+
+                if (previousElement?.name === viewItem.parent.name) {
+                    conversionApi.writer.mergeContainers(conversionApi.writer.createPositionAfter(previousElement));
+                }
+
+                if (nextElement?.name === viewItem.parent.name) {
+                    conversionApi.writer.mergeContainers(conversionApi.writer.createPositionBefore(nextElement));
+                }
+            });
+        });
+
+        this.editor.conversion.for('upcast').add((dispatcher) => {
+            dispatcher.on('element:li', (event, data, conversionApi) => {
+                const listParent = data.viewItem.parent;
+                const listItem = data.modelRange.start.nodeAfter || data.modelRange.end.nodeBefore;
```

Comment:
```suggestion
                const listItem = data.modelRange.start.nodeAfter ?? data.modelRange.end.nodeBefore;
```
in theory, but that's huuuuge nitpick:p

Reply:
I dont agree its nitpick.
Lets say nodeAfter is false or 0.
I recently had bug where ?? broke code since '' was not treated as falsy value.